lib/Controller/Library.php

```diff
@@ -330,6 +330,84 @@ function displayPage()
         ]);
     }
 
+    /**
+     * Set Enable Stats Collection of a media
+     * @param int $mediaId
+     *
+     * @SWG\Put(
+     *  path="/library/setenablestat/{mediaId}",
+     *  operationId="mediaSetEnableStat",
+     *  tags={"library"},
+     *  summary="Enable Stats Collection",
+     *  description="Set Enable Stats Collection? to use for the collection of Proof of Play statistics for a media.",
+     *  @SWG\Parameter(
+     *      name="mediaId",
+     *      in="path",
+     *      description="The Media ID",
+     *      type="integer",
+     *      required=true
+     *   ),
+     *  @SWG\Parameter(
+     *      name="enableStat",
+     *      in="formData",
+     *      description="The option to enable the collection of Media Proof of Play statistics,
+     *      type="string",
+     *      required=true
+     *   ),
+     *  @SWG\Response(
+     *      response=204,
+     *      description="successful operation"
+     *  )
+     * )
+     *
+     * @throws XiboException
+     */
+
+    function setEnableStat($mediaId)
+    {
+        // Get the Media
+        $media = $this->mediaFactory->getById($mediaId);
+
+        // Check Permissions
+        if (!$this->getUser()->checkViewable($media))
+            throw new AccessDeniedException();
+
+        $enableStat = $this->getSanitizer()->getString('enableStat');
+
+        $media->enableStat = $enableStat;
+        $media->save();
+
+        // Return
+        $this->getState()->hydrate([
+            'httpStatus' => 204,
+            'message' => sprintf(__('For Media %s Enable Stats Collection is set to %s'), $media->name, $media->enableStat)
```

Comment:
Did you want to have your On/Off string here too (it might be nice instead of showing 0 / 1)

Reply:
This is already a string i.e., on/off/inherit

Reply:
`$media->enableStat` is an integer?

Reply:
$media->enableStat is a string

Reply:
as it is a string, I have done as below:

> sprintf(__('For Media %s Enable Stats Collection is set to %s'), $media->name, __($media->enableStat))

Reply:
ah I understand! yes that is perfect